Jakarta-ized platform/TransactionManagement.adoc

b6arat6 opened the pull request:
Signed-off-by: Bharath Kandasamy <bh4r4th@gmail.com>

b6arat6 commented:
@tomjenkinson Fixed the line ending issue.

xstefank commented on specification/src/main/asciidoc/platform/TransactionManagement.adoc:
```diff
@@ -54,27 +54,27 @@ transaction capabilities to support such a scenario.
 Web components may demarcate transactions using
-the _javax.transaction.UserTransaction_ interface or transactional
-interceptors, which are defined in the JTA specification. They may
+the _jakarta.transaction.UserTransaction_ interface or transactional
```
this shouldn't be changed in Jakarta EE 8 because it should be the same as Java EE 8.

tomjenkinson replied:
+1, good catch!

tomjenkinson replied:
This same issue appears a few times in the PR ("jakarta.t")

tomjenkinson commented on specification/src/main/asciidoc/platform/TransactionManagement.adoc:
```diff
@@ -54,27 +54,27 @@ transaction capabilities to support such a scenario.
-the _javax.transaction.UserTransaction_ interface or transactional
-interceptors, which are defined in the JTA specification. They may
+the _jakarta.transaction.UserTransaction_ interface or transactional
+interceptors, which are defined in the Jakarta Transaction specification. They may
```
This is a specific example of my earlier review comment, i.e. "Jakarta Transaction" -> "Jakarta Transactions"

kwsutter commented on specification/src/main/asciidoc/platform/TransactionManagement.adoc:
```diff
@@ -84,46 +84,46 @@ required to use Container Managed Transactions).
-to web components under the name _java:comp/UserTransaction_ .
-* The Java EE platform must provide classes
-that implement the transactional interceptors, as defined by the JTA
+to web components under the name _jakarta:comp/UserTransaction_ .
```
still java:comp/UserTransaction

kwsutter commented on specification/src/main/asciidoc/platform/TransactionManagement.adoc:
```diff
@@ -240,19 +240,19 @@ access a database, invoke an enterprise bean that accesses the same
 === [[a520]]Transactional JMS Support
 
-A Java EE product must support a JMS provider
+A Jakarta EE product must support a JMS provider
```
Need to change JMS to Jakarta Messaging

b6arat6 replied:
Resolved

kwsutter commented on specification/src/main/asciidoc/platform/TransactionManagement.adoc:
```diff
@@ -240,19 +240,19 @@ access a database, invoke an enterprise bean that accesses the same
-A Java EE product must support a JMS provider
+A Jakarta EE product must support a JMS provider
 as a transactional resource manager. The platform must enable
 transactional JMS access from servlets, JSP pages, and enterprise beans.
```
Change JSP to "server pages" (generalize the use)

b6arat6 replied:
Resolved

kwsutter commented on specification/src/main/asciidoc/platform/TransactionManagement.adoc:
```diff
@@ -264,7 +264,7 @@ same transaction, and, finally, commit the transaction.
+A Jakarta EE product must support resource
 adapters that use _XATransaction_ mode as transactional resource
 managers. The platform must enable transactional access to the resource
 adapter from servlets, JSP pages, and enterprise beans.
```
change JSP to server pages (generalize)

kwsutter commented on specification/src/main/asciidoc/platform/TransactionManagement.adoc:
```diff
@@ -278,27 +278,27 @@ finally, commit the transaction.
+We recommend that Jakarta EE Product Providers use
 the IIOP transaction propagation protocol defined by OMG and described
 in the OTS specification (and implemented by the Java Transaction
 Service), for transaction interoperability when using the EJB
```
Probably need to modify EJB to Jakarta Enterprise Beans

b6arat6 replied:
Resolved

kwsutter commented on specification/src/main/asciidoc/platform/TransactionManagement.adoc:
```diff
@@ -278,27 +278,27 @@ finally, commit the transaction.
-requires all Java EE products to support the
+requires all Jakarta EE products to support the
 _javax.transaction.xa.XAResource_ interface for performing two-phase
 commit operations on JDBC drivers that support the JTA XA APIs. This
```
Change JTA to Jakarta Transaction

kwsutter commented on specification/src/main/asciidoc/platform/TransactionManagement.adoc:
```diff
@@ -410,29 +410,29 @@ link:#a516[See Transactional JDBC™
 link:#a520[See Transactional JMS
 Support]” may impose some restrictions on a Deployer’s configuration of
-an application’s JDBC and JMS resources. Java EE Product Providers may
+an application’s JDBC and JMS resources. Jakarta EE Product Providers may
```
JMS should be changed to Jakarta Messaging

kwsutter commented on specification/src/main/asciidoc/platform/TransactionManagement.adoc:
```diff
@@ -410,29 +410,29 @@ link:#a516[See Transactional JDBC™
 Technology Support]” and the JMS transaction requirements in
```
JMS should be Jakarta Messaging

kwsutter commented on specification/src/main/asciidoc/platform/TransactionManagement.adoc:
```diff
@@ -410,29 +410,29 @@ link:#a516[See Transactional JDBC™
-A Java EE Product Provider may make the same
+A Jakarta EE Product Provider may make the same
 restrictions as above, resulting in a transaction being restricted to a
 single JMS resource manager and user identity.
```
JMS change to Jakarta Messaging

kwsutter commented on specification/src/main/asciidoc/platform/TransactionManagement.adoc:
```diff
@@ -441,13 +441,13 @@ to include JDBC and JMS access in a single global transaction must not
 
 Although these restrictions are allowed, it is
-recommended that Java EE Product Providers support JDBC and JMS resource
+recommended that Jakarta EE Product Providers support JDBC and JMS resource
```
Jakarta Messaging

b6arat6 commented:
Thank you all for reviewing the changes. I will incorporate the changes and
submit for review.

Bharath

On Fri, Jul 5, 2019, 6:07 PM Kevin Sutter <notifications@github.com> wrote:

> *@kwsutter* requested changes on this pull request.
>
> Thanks for the PR! Just a few changes to make it ready for approval.,
> ------------------------------
>
> In specification/src/main/asciidoc/platform/TransactionManagement.adoc
> <https://github.com/eclipse-ee4j/jakartaee-platform/pull/50#discussion_r300815021>
> :
>
> > @@ -1,18 +1,18 @@
>
>  == Transaction Management
>
>
>
>  This chapter describes the required Java™
>
> -Platform, Enterprise Edition (Java EE) transaction management and
>
> +Platform, Enterprise Edition (Jakarta EE) transaction management and
>
>
> This should indicate "..the required Jakarta Enterprise Edition (Jakarta
> EE) transaction.." The use of "Platform" is not correct in this context.
> ------------------------------
>
> In specification/src/main/asciidoc/platform/TransactionManagement.adoc
> <https://github.com/eclipse-ee4j/jakartaee-platform/pull/50#discussion_r300815059>
> :
>
> > @@ -22,10 +22,10 @@ adapters specifying the _XATransaction_ transaction level
>
>
>
>  === Overview
>
>
>
> -A Java EE Product that includes both a servlet
>
> +A Jakarta EE Product that includes both a servlet
>
>  container and an EJB container must support a transactional application
>
>
> genericize the use of EJB -- "both a servlet container and an enterprise
> bean container.."
> ------------------------------
>
> In specification/src/main/asciidoc/platform/TransactionManagement.adoc
> <https://github.com/eclipse-ee4j/jakartaee-platform/pull/50#discussion_r300815098>
> :
>
> >  Providers.
>
>
>
>  === Web Components
>
>
>
>  Web components may demarcate transactions using
>
> -the _javax.transaction.UserTransaction_ interface or transactional
>
> -interceptors, which are defined in the JTA specification. They may
>
> +the _jakarta.transaction.UserTransaction_ interface or transactional
>
>
> @xstefank <https://github.com/xstefank> is correct. No package name
> changes in Jakarta EE 8. Thanks.,
> ------------------------------
>
> In specification/src/main/asciidoc/platform/TransactionManagement.adoc
> <https://github.com/eclipse-ee4j/jakartaee-platform/pull/50#discussion_r300815118>
> :
>
> > @@ -84,46 +84,46 @@ required to use Container Managed Transactions).
>
>
>
>  Web application event listeners and upgrade
>
>  handlers must not demarcate transactions using the
>
> -_javax.transaction.UserTransaction_ interface or transactional
>
> +_jakarta.transaction.UserTransaction_ interface or transactional
>
>
> again, don't change to jakarta.* yet...
> ------------------------------
>
> In specification/src/main/asciidoc/platform/TransactionManagement.adoc
> <https://github.com/eclipse-ee4j/jakartaee-platform/pull/50#discussion_r300815137>
> :
>
> >  requirements:
>
>
>
> -* The Java EE platform must provide an object
>
> -implementing the _javax.transaction.UserTransaction_ interface to all
>
> +* The Jakarta EE platform must provide an object
>
> +implementing the _jakarta.transaction.UserTransaction_ interface to all
>
>
> javax.transaction for jakarta ee 8
> ------------------------------
>
> In specification/src/main/asciidoc/platform/TransactionManagement.adoc
> <https://github.com/eclipse-ee4j/jakartaee-platform/pull/50#discussion_r300815147>
> :
>
> >  web components. The platform must publish the _UserTransaction_ object
>
>  in the Java™ Naming and Directory Interface (JNDI) name space available
>
> -to web components under the name _java:comp/UserTransaction_ .
>
> -* The Java EE platform must provide classes
>
> -that implement the transactional interceptors, as defined by the JTA
>
> +to web components under the name _jakarta:comp/UserTransaction_ .
>
>
> still java:comp/UserTransaction
> ------------------------------
>
> In specification/src/main/asciidoc/platform/TransactionManagement.adoc
> <https://github.com/eclipse-ee4j/jakartaee-platform/pull/50#discussion_r300815193>
> :
>
> >  drivers through product-specific means, although normally such JDBC
>
>  drivers would be delivered as resource adapters using the Connector API.
>
>
>
>  === [[a520]]Transactional JMS Support
>
>
>
> -A Java EE product must support a JMS provider
>
> +A Jakarta EE product must support a JMS provider
>
>
> Need to change JMS to Jakarta Messaging
> ------------------------------
>
> In specification/src/main/asciidoc/platform/TransactionManagement.adoc
> <https://github.com/eclipse-ee4j/jakartaee-platform/pull/50#discussion_r300815208>
> :
>
> >  drivers through product-specific means, although normally such JDBC
>
>  drivers would be delivered as resource adapters using the Connector API.
>
>
>
>  === [[a520]]Transactional JMS Support
>
>
>
> -A Java EE product must support a JMS provider
>
> +A Jakarta EE product must support a JMS provider
>
>  as a transactional resource manager. The platform must enable
>
>  transactional JMS access from servlets, JSP pages, and enterprise beans.
>
>
> Change JSP to "server pages" (generalize the use)
> ------------------------------
>
> In specification/src/main/asciidoc/platform/TransactionManagement.adoc
> <https://github.com/eclipse-ee4j/jakartaee-platform/pull/50#discussion_r300815218>
> :
>
> > @@ -264,7 +264,7 @@ same transaction, and, finally, commit the transaction.
>
>
>
>  === Transactional Resource Adapter (Connector) Support
>
>
>
> -A Java EE product must support resource
>
> +A Jakarta EE product must support resource
>
>  adapters that use _XATransaction_ mode as transactional resource
>
>  managers. The platform must enable transactional access to the resource
>
>  adapter from servlets, JSP pages, and enterprise beans.
>
>
> change JSP to server pages (generalize)
> ------------------------------
>
> In specification/src/main/asciidoc/platform/TransactionManagement.adoc
> <https://github.com/eclipse-ee4j/jakartaee-platform/pull/50#discussion_r300815241>
> :
>
> >  multiple Product Providers.
>
>
>
> -We recommend that Java EE Product Providers use
>
> +We recommend that Jakarta EE Product Providers use
>
>  the IIOP transaction propagation protocol defined by OMG and described
>
>  in the OTS specification (and implemented by the Java Transaction
>
>  Service), for transaction interoperability when using the EJB
>
>
> Probably need to modify EJB to Jakarta Enterprise Beans
> ------------------------------
>
> In specification/src/main/asciidoc/platform/TransactionManagement.adoc
> <https://github.com/eclipse-ee4j/jakartaee-platform/pull/50#discussion_r300815261>
> :
>
> >  products to support the _javax.transaction.xa.XAResource_ interface, as
>
>  specified in the Connector specification. This specification also
>
> -requires all Java EE products to support the
>
> +requires all Jakarta EE products to support the
>
>  _javax.transaction.xa.XAResource_ interface for performing two-phase
>
>  commit operations on JDBC drivers that support the JTA XA APIs. This
>
>
> Change JTA to Jakarta Transaction
> ------------------------------
>
> In specification/src/main/asciidoc/platform/TransactionManagement.adoc
> <https://github.com/eclipse-ee4j/jakartaee-platform/pull/50#discussion_r300815283>
> :
>
> > @@ -410,29 +410,29 @@ link:#a516[See Transactional JDBC™
>
>  Technology Support]” and the JMS transaction requirements in
>
>  link:#a520[See Transactional JMS
>
>  Support]” may impose some restrictions on a Deployer’s configuration of
>
> -an application’s JDBC and JMS resources. Java EE Product Providers may
>
> +an application’s JDBC and JMS resources. Jakarta EE Product Providers may
>
>
> JMS should be changed to Jakarta Messaging
> ------------------------------
>
> In specification/src/main/asciidoc/platform/TransactionManagement.adoc
> <https://github.com/eclipse-ee4j/jakartaee-platform/pull/50#discussion_r300815290>
> :
>
> > @@ -410,29 +410,29 @@ link:#a516[See Transactional JDBC™
>
>  Technology Support]” and the JMS transaction requirements in
>
>
> JMS should be Jakarta Messaging
> ------------------------------
>
> In specification/src/main/asciidoc/platform/TransactionManagement.adoc
> <https://github.com/eclipse-ee4j/jakartaee-platform/pull/50#discussion_r300815298>
> :
>
> >  required to support transactions where more than one JDBC identity is
>
>  used. Specifically, this means that transactions that require the use of
>
>  more than one JDBC security identity (which can be done explicitly via
>
>  component provided user name and password) may not be portable.
>
>
>
> -A Java EE Product Provider may make the same
>
> +A Jakarta EE Product Provider may make the same
>
>  restrictions as above, resulting in a transaction being restricted to a
>
>  single JMS resource manager and user identity.
>
>
> JMS change to Jakarta Messaging
> ------------------------------
>
> In specification/src/main/asciidoc/platform/TransactionManagement.adoc
> <https://github.com/eclipse-ee4j/jakartaee-platform/pull/50#discussion_r300815307>
> :
>
> > @@ -441,13 +441,13 @@ to include JDBC and JMS access in a single global transaction must not
>
>  mark the corresponding transactional resources as “unshareable”.
>
>
>
>  Although these restrictions are allowed, it is
>
> -recommended that Java EE Product Providers support JDBC and JMS resource
>
> +recommended that Jakarta EE Product Providers support JDBC and JMS resource
>
>
> Jakarta Messaging
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/eclipse-ee4j/jakartaee-platform/pull/50?email_source=notifications&email_token=AETOVHBHHUPONNKBHVYRKVTP57V3RA5CNFSM4H6LBKIKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOB5UXZ3A#pullrequestreview-258571500>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AETOVHFSPK3KXW35PXJ56SDP57V3RANCNFSM4H6LBKIA>
> .
>

smillidge commented on specification/src/main/asciidoc/platform/TransactionManagement.adoc:
```diff
@@ -209,26 +208,26 @@ components (using the _forward_ or _include_ methods) should not store
-requirements are described above and in the EJB specification.
+requirements are described above and in the Jakarta Enterprise Beans specification.
 
 === Enterprise JavaBeans™ Components
```
The title should be `Jakarta Enterprise Beans`

smillidge commented on specification/src/main/asciidoc/platform/TransactionManagement.adoc:
```diff
@@ -240,34 +239,34 @@ access a database, invoke an enterprise bean that accesses the same
+A Jakarta EE product must provide a transaction
 manager that is capable of coordinating two-phase commit operations
 across multiple XA-capable JDBC databases. If a JDBC driver supports the
 Java Transaction API’s XA interfaces (in the _javax.transaction.xa_
```
This should be `Jakarta Transactions`

bshannon replied:
The javax.transaction.xa package remains with Java SE and is defined by the JTA XA spec,
not the Jakarta Transactions spec.

b6arat6 commented:
@kwsutter @smillidge @tomjenkinson Incorporated all the review comments. Please review.

tomjenkinson commented on specification/src/main/asciidoc/platform/TransactionManagement.adoc:
```diff
@@ -92,38 +91,38 @@ objects.
-to web components under the name _java:comp/UserTransaction_ .
-* The Java EE platform must provide classes
-that implement the transactional interceptors, as defined by the JTA
+to web components under the name java:comp/UserTransaction_ .
```
I wonder what this trailing "_" is about? maybe it should be "_java" still to do something like emphasis?

kwsutter replied:
Correct, @tomjenkinson.  He missed the underscore before the "_java"...  Thanks!

tomjenkinson commented on specification/src/main/asciidoc/platform/TransactionManagement.adoc:
```diff
@@ -92,38 +91,38 @@ objects.
 * If a web component creates a thread, the Java
 EE platform must ensure that the newly created thread is not associated
-with any JTA transaction.
+with any Jakarta transactions.
```
"Jakarta Transaction transaction" maybe?

tomjenkinson commented on specification/src/main/asciidoc/platform/TransactionManagement.adoc:
```diff
@@ -176,10 +175,10 @@ between the use of transactional resources and threads. To ensure
 and the web container must support at least these usages.
 
-* JTA transactions should be started and
+* Jakarta Transactions should be started and
```
"Jakarta Transactions transactions" I think

bshannon replied:
Or just "Transactions"?  Or "Transactions defined by the Jakarta Transactions specification"?

"Jakarta Transactions transactions" is just too clumsy...

b6arat6 replied:
Adopted to referring "Transactions defined by the Jakarta Transactions specification".

tomjenkinson commented on specification/src/main/asciidoc/platform/TransactionManagement.adoc:
```diff
@@ -176,10 +175,10 @@ between the use of transactional resources and threads. To ensure
 completed in the thread in which the _service_ method is called.
 Additional threads that are created for any purpose should not attempt
-to start JTA transactions.
+to start Jakarta Transactions.
```
"Jakarta Transactions transactions" I think

tomjenkinson commented on specification/src/main/asciidoc/platform/TransactionManagement.adoc:
```diff
@@ -278,37 +277,36 @@ finally, commit the transaction.
 the IIOP transaction propagation protocol defined by OMG and described
-in the OTS specification (and implemented by the Java Transaction
-Service), for transaction interoperability when using the EJB
+in the OTS specification (and implemented by the Jakarta Transactions
```
@bshannon how is JTS being handled in the transition to Jakarta?

bshannon replied:
JTS was never a JCP spec.  My understanding is that it is "just" a Java mapping to OTS.
The entire implementation is included in the GlassFish ORB.

The simplest thing is probably just to remove the reference to "Java Transaction Service".
Definitely do NOT change it to "Jakarta Transaction Service".

b6arat6 replied:
Removed reference to Java Transaction Service.

tomjenkinson commented on specification/src/main/asciidoc/platform/TransactionManagement.adoc:
```diff
@@ -278,37 +277,36 @@ finally, commit the transaction.
 _javax.transaction.xa.XAResource_ interface for performing two-phase
-commit operations on JDBC drivers that support the JTA XA APIs. This
-specification does not require that JDBC drivers or JMS providers use
+commit operations on JDBC drivers that support the Jakarta Transactions XA APIs. This
```
I think XAResource is part of Java SE now? https://docs.oracle.com/javase/8/docs/api/javax/transaction/xa/XAResource.html

bshannon replied:
Yes, the JTA XA APIs are not part of the Jakarta Transactions spec.

tomjenkinson commented on specification/src/main/asciidoc/platform/TransactionManagement.adoc:
```diff
@@ -92,38 +91,38 @@ objects.
 specification.
 * If a web component invokes an enterprise bean
-from a thread associated with a JTA transaction, the Java EE platform
+from a thread associated with a Jakarta Transactions, the Jakarta EE platform
```
"Jakarta Transactions transactions" I think

tomjenkinson commented on specification/src/main/asciidoc/platform/TransactionManagement.adoc:
```diff
@@ -92,38 +91,38 @@ objects.
-resource manager from a thread associated with a JTA transaction, the
-Java EE platform must ensure that the resource access is included as
-part of the JTA transaction.
+resource manager from a thread associated with a Jakarta Transactions, the
```
"Jakarta Transactions transactions" I think

tomjenkinson commented on specification/src/main/asciidoc/platform/TransactionManagement.adoc:
```diff
@@ -92,38 +91,38 @@ objects.
-part of the JTA transaction.
+resource manager from a thread associated with a Jakarta Transactions, the
+Jakarta EE platform must ensure that the resource access is included as
+part of the Jakarta Transactions.
```
"Jakarta Transactions transactions" I think

b6arat6 commented:
Incorporated the review comments.

Changes:
1. Removed reference to Java Transaction Service
2. Retained reference to JTA XA API's
3. Transaction/Transactions defined by the Jakarta Transactions spec
4. Broke lengthy sentences

tomjenkinson commented on specification/src/main/asciidoc/platform/TransactionManagement.adoc:
```diff
@@ -1,43 +1,42 @@
-=== [[a475]]Servlets/JSP Pages Accessing Enterprise Beans
+=== [[a475]]Servlets/Server Pages Accessing Enterprise Beans
 
 image:Platform_Spec-8.png[image]
```
When I viewed this on https://github.com/eclipse-ee4j/jakartaee-platform/blob/710bc5e80e163b33f3acf47f7a089830d16215d1/specification/src/main/asciidoc/platform/TransactionManagement.adoc#a493 it was 404'ing?

b6arat6 replied:
I am able to view https://github.com/eclipse-ee4j/jakartaee-platform/blob/710bc5e80e163b33f3acf47f7a089830d16215d1/specification/src/main/asciidoc/platform/TransactionManagement.adoc#a493 without any issues. Please recheck and let me know.

tomjenkinson replied:
Is not working for me: https://github.com/eclipse-ee4j/jakartaee-platform/blob/710bc5e80e163b33f3acf47f7a089830d16215d1/specification/src/main/asciidoc/platform/Platform_Spec-8.png - the 404 I was referring to was the image in there, not the entire document

tomjenkinson commented on specification/src/main/asciidoc/platform/TransactionManagement.adoc:
```diff
@@ -92,38 +91,40 @@ objects.
-Invocations of enterprise beans in another Java EE product instance (for
-example, using the EJB interoperability protocol) need not propagate the
-transaction context. See the EJB specification for details.
+Jakarta EE product instancelink:#a3649[2] as the invoking component.
```
I don't think this anchor exists?

b6arat6 replied:
Yes, this anchor does not exist. I removed it now.

tomjenkinson commented on specification/src/main/asciidoc/platform/TransactionManagement.adoc:
```diff
@@ -386,73 +389,74 @@ marked unshareable. If a connection is not marked unshareable, it must
-Application Client XML Schema]”, the EJB specification, and the Servlet
-specification provide descriptions of the deployment descriptor element.
+is provided. link:#a3399[See Jakarta EE
+Application Client XML Schema]”, the Jakarta Enterprise Beans specification, and
```
the ” seems to be superfluous?

tomjenkinson commented on specification/src/main/asciidoc/platform/TransactionManagement.adoc:
```diff
@@ -386,73 +389,74 @@ marked unshareable. If a connection is not marked unshareable, it must
 link:#a516[See Transactional JDBC™
-Technology Support]” and the JMS transaction requirements in
-link:#a520[See Transactional JMS
+Technology Support]” and the Jakarta Messaging transaction requirements in
```
the ” seems to be superfluous?

tomjenkinson commented on specification/src/main/asciidoc/platform/TransactionManagement.adoc:
```diff
@@ -386,73 +389,74 @@ marked unshareable. If a connection is not marked unshareable, it must
-link:#a520[See Transactional JMS
+Technology Support]” and the Jakarta Messaging transaction requirements in
+link:#a520[See Transactional Jakarta Messaging
 Support]” may impose some restrictions on a Deployer’s configuration of
```
the ” seems to be superfluous?

tomjenkinson commented on specification/src/main/asciidoc/platform/TransactionManagement.adoc:
```diff
@@ -463,7 +467,7 @@ Technology Support].”
 === System Administration Tools
```
the ” seems to be superfluous? (two lines up related to JDBC

tomjenkinson commented on specification/src/main/asciidoc/platform/TransactionManagement.adoc:
```diff
@@ -139,7 +140,7 @@ propagated to the target servlet or page.
+context must be propagated to the called servlet or server pages.
 
 === Transactions in Web Component Life Cycles
 
```
About six lines below there is a superfluous ” after:
> As specified above in See Transaction Non-Requirements,”

tomjenkinson commented on specification/src/main/asciidoc/platform/TransactionManagement.adoc:
```diff
@@ -1,43 +1,42 @@
-=== [[a475]]Servlets/JSP Pages Accessing Enterprise Beans
+=== [[a475]]Servlets/Server Pages Accessing Enterprise Beans
 
 image:Platform_Spec-8.png[image]
```
Is not working for me: https://github.com/eclipse-ee4j/jakartaee-platform/blob/710bc5e80e163b33f3acf47f7a089830d16215d1/specification/src/main/asciidoc/platform/Platform_Spec-8.png - the 404 I was referring to was the image in there, not the entire document

b6arat6 replied:
Yes, I also get 404. The link in master (https://github.com/eclipse-ee4j/jakartaee-platform/blob/master/specification/src/main/asciidoc/platform/Platform_Spec-8.png) throws 404 too. What should we do?

tomjenkinson replied:
My guess is it is meant to be linked to https://github.com/eclipse-ee4j/jakartaee-platform/blob/master/specification/src/main/asciidoc/images/Platform_Spec-8.png - perhaps it does that during the build or maybe the URL needs images/ in the path?

b6arat6 replied:
image:images/Platform_Spec-8.png[image] fixed the image issue

tomjenkinson commented:
It is still not looking right sorry - thanks for your patience!
If you look at https://github.com/eclipse-ee4j/jakartaee-platform/blob/d9896d669c85638e6630ab4cbb3c3c199fbd0cab/specification/src/main/asciidoc/platform/TransactionManagement.adoc you can see the image is linking to https://github.com/eclipse-ee4j/jakartaee-platform/blob/d9896d669c85638e6630ab4cbb3c3c199fbd0cab/specification/src/main/asciidoc/platform/images/Platform_Spec-8.png but it should be https://github.com/eclipse-ee4j/jakartaee-platform/blob/d9896d669c85638e6630ab4cbb3c3c199fbd0cab/specification/src/main/asciidoc/images/Platform_Spec-8.png

b6arat6 commented:
I am not sure why Eclipse ECA Validation has failed because I added the footer and I signed ECA already. I could not verify the ECA now because my account is locked, when I tried to login with an incorrect password for > 5 times. I will verify the ECA later.

kwsutter commented:
Thank you for the PR @b6arat6 and the great reviewing @tomjenkinson!    I will try to find out what's wrong with the ECA check...  If you click on the Details item next to that error, it indicates that it is still missing the "signed-off-by" footer.  In the mean time, you might want to post to the incubator mailing list to see if they can help out (https://accounts.eclipse.org/mailing-list/incubation).

kwsutter commented:
@b6arat6 It looks like your last three commits starting with [this one](https://github.com/eclipse-ee4j/jakartaee-platform/pull/50/commits/75964ed65ea6cf7d32792c24eb4825a66cd863e1) have an issue with not being verified.  The first several commits are all clear, but the last few commits have the issue.  Something for you to look at...

b6arat6 commented:
@kwsutter I forgot to add the footer to one of my last 3 commits. I have sent an email to incubation@eclipse.org for the commit validation failure and started discussing the issue with them.

kwsutter commented:
> @kwsutter I forgot to add the footer to one of my last 3 commits. I have sent an email to [incubation@eclipse.org](mailto:incubation@eclipse.org) for the commit validation failure and started discussing the issue with them.

If you don't get assistance tomorrow morning via your incubation discussion, then let me know and we'll try another avenue...  It would be good to get this PR done and out of the way.  Thanks.

b6arat6 commented:
> Thanks @b6arat6 . I think you addressed all my comments except one which is that if you look at https://github.com/eclipse-ee4j/jakartaee-platform/blob/540e45779515dd8f32464c08bda632d4c5ae7e89/specification/src/main/asciidoc/platform/TransactionManagement.adoc and search it for a double quote (") you will see 5 spurious examples (there are 7 on the page but two are around the word unshareable which seems valid to me. Would you be OK to address those please?

@tomjenkinson You are absolutely right! Seriously, how do you catch them? :) I will fix them in a few hours, when I reach home, after work.

b6arat6 commented:
> > @kwsutter I forgot to add the footer to one of my last 3 commits. I have sent an email to [incubation@eclipse.org](mailto:incubation@eclipse.org) for the commit validation failure and started discussing the issue with them.
> 
> If you don't get assistance tomorrow morning via your incubation discussion, then let me know and we'll try another avenue... It would be good to get this PR done and out of the way. Thanks.

I was asked to edit the commit using "git rebase" to include the footer. I will try it in a few hours tonight.

b6arat6 commented:
> I sure wish the text hadn't all been reformatted.
> It makes it really difficult to determine what has changed
> and I don't have time to wade through it all right now. :-(

I understand and I will take care of it next time.

tomjenkinson commented:
@b6arat6 I think that the formatting is still off sorry - you can see for yourself on https://github.com/eclipse-ee4j/jakartaee-platform/pull/50/files :'(

I think you might be able to revert to 540e45779515dd8f32464c08bda632d4c5ae7e89 and then do the quote removal again (similar to b52d0c50d254bade8796a60436ec7888724a28d5 but I doubt it can be cherry-picked because the underlying structure of the text files is probably aligned differently).
